core/modules/path/path.module

```diff
@@ -405,8 +732,36 @@ function path_entity_update(Entity $entity) {
     $langcode = isset($entity->langcode) ? $entity->langcode : LANGUAGE_NONE;
     $uri = $entity->uri();
     $path = $entity->path;
+    $path_original = $entity->original->path;
     $path['source'] = $uri['path'];
 
+    //save a redirect if the path is changed on node save.
+    if (!config_get('redirect.settings', 'auto_redirect')) {
+      return;
+    }
+    elseif (isset($path['redirect']) && !$path['redirect']) {
+      return;
+    }
+
+    if (
+        // !empty($path_original['pid']) &&
+        // $path_original['pid'] == $path['pid'] &&
+        // $path_original['alias'] != $path['alias']
+        $entity->title != $entity->original->title
```

Comment:
Is this condition correct? It seems like redirects could be created on conditions other than just the title changing. Seems like everything that is commented out would be correct and the title check should be removed.